src/utils/core_app.rs

```diff
@@ -0,0 +1,133 @@
+pub use crate::types::happ::{HappPreferences, PresentedHappBundle, ServiceloggerHappPreferences};
+use crate::types::transaction::PendingTransaction;
+use anyhow::Result;
+use holochain_types::dna::{ActionHashB64, AgentPubKey};
+use holochain_types::prelude::{ExternIO, FunctionName, ZomeName};
+use hpos_hc_connect::app_connection::CoreAppRoleName;
+use hpos_hc_connect::hha_agent::HHAAgent;
+use hpos_hc_connect::hha_types::HappAndHost;
+use tracing::trace;
+
+pub async fn get_host_preferences(core_app_client: &mut HHAAgent) -> Result<HappPreferences> {
```

Comment:
Suggestion: All these zomeCalls could be added to the `HHAAgent` impl here: https://github.com/Holo-Host/hpos-service-crates/blob/main/crates/hpos_connect_hc/src/hha_agent.rs#L56-L89

Reply:
Done.  Can you take a look at the dependent pr: https://github.com/Holo-Host/hpos-service-crates/pull/191